src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubConfiguration.java

```diff
@@ -74,4 +120,97 @@ public void setEndpoints(@CheckForNull List<Endpoint> endpoints) {
         save();
     }
 
+    /**
+     * Adds an endpoint.
+     *
+     * @param endpoint the endpoint to add.
+     * @return {@code true} if the list of endpoints was modified
+     */
+    public synchronized boolean addEndpoint(@NonNull Endpoint endpoint) {
+        if (StringUtils.isBlank(endpoint.getApiUri())) {
+            return false;
+        }
+        List<Endpoint> endpoints = new ArrayList<>(getEndpoints());
+        for (Endpoint ep : endpoints) {
+            if (ep.getApiUri().equals(endpoint.getApiUri())) {
+                return false;
+            }
+        }
+        endpoints.add(endpoint);
+        setEndpoints(endpoints);
+        return true;
+    }
+
+    /**
+     * Updates an existing endpoint (or adds if missing).
+     *
+     * @param endpoint the endpoint to update.
+     */
+    public synchronized void updateEndpoint(@NonNull Endpoint endpoint) {
+        if (StringUtils.isBlank(endpoint.getApiUri())) {
+            return;
+        }
+        List<Endpoint> endpoints = new ArrayList<>(getEndpoints());
+        boolean found = false;
+        for (int i = 0; i < endpoints.size(); i++) {
+            Endpoint ep = endpoints.get(i);
+            if (ep.getApiUri().equals(endpoint.getApiUri())) {
+                endpoints.set(i, endpoint);
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            endpoints.add(endpoint);
+        }
+        setEndpoints(endpoints);
+    }
+
+    /**
+     * Removes an endpoint.
+     *
+     * @param endpoint the endpoint to remove.
+     * @return {@code true} if the list of endpoints was modified
+     */
+    public boolean removeEndpoint(@NonNull Endpoint endpoint) {
+        return removeEndpoint(endpoint.getApiUri());
+    }
+
+    /**
+     * Removes an endpoint.
+     *
+     * @param apiUri the API URI to remove.
+     * @return {@code true} if the list of endpoints was modified
+     */
+    public synchronized boolean removeEndpoint(@CheckForNull String apiUri) {
+        apiUri = normalizeApiUri(apiUri);
+        boolean modified = false;
+        List<Endpoint> endpoints = new ArrayList<>(getEndpoints());
+        for (Iterator<Endpoint> iterator = endpoints.iterator(); iterator.hasNext(); ) {
+            if (apiUri.equals(iterator.next().getApiUri())) {
```

Comment:
no, we strip out nulls from the list of endpoints. There are no nulls to worry about here